src/Entity/EthereumServer.php

```diff
@@ -105,13 +107,17 @@ public function isDefaultServer() {
    * {@inheritdoc}
    */
   public function validateConnection() {
-    $return = ['error' => FALSE, 'message' => ''];
+    $return = [
+      'error' => FALSE,
+      'message' => 'Server ' . $this->id . ' (' . $this->url .') is up and on listening on Ethereum network id ' . $this->network_id
+    ];
     try {
-      /** @var \Ethereum\Ethereum $client */
-      $client = \Drupal::service('ethereum.client_factory')->get($this->getUrl());
+      /** @var \Ethereum\Ethereum $web3 */
+       $web3 = new Ethereum($this->url);
+      // @todo How to load the \Drupal::service('ethereum.client') with a updated url?
```

Comment:
The previous code was already handling that..

Reply:
ok. Reverted to your code.